tests/Resources/Page/ODH/JupyterHub/JupyterHubSpawner.robot

```diff
@@ -132,20 +132,24 @@ Spawn Notebook With Arguments
    [Documentation]  Selects required settings and spawns a notebook pod. If it fails due to timeout or other issue
    ...              It will try again ${retries} times (Default: 1). Environment variables can be passed in as kwargs
    ...              By creating a dictionary beforehand, e.g. &{test-dict}  Create Dictionary  name=robot  password=secret
-   [Arguments]  ${retries}=1  ${image}=s2i-generic-data-science-notebook  ${size}=Small  ${spawner_timeout}=600 seconds  &{envs}
+   [Arguments]  ${retries}=1  ${image}=s2i-generic-data-science-notebook  ${size}=Small  ${spawner_timeout}=600 seconds  ${refresh}=${False}  &{envs}
    FOR  ${index}  IN RANGE  0  1+${retries}
       ${spawner_ready} =    Run Keyword and Return Status    Wait Until JupyterHub Spawner Is Ready
       IF  ${spawner_ready}==True
          Select Notebook Image  ${image}
          Select Container Size  ${size}
+         IF   ${refresh}
+              Reload Page
+              Capture Page Screenshot    reload.png
+         END
          IF  &{envs}
             Remove All Spawner Environment Variables
             FOR  ${key}  ${value}  IN  &{envs}[envs]
                Sleep  1
                Add Spawner Environment Variable  ${key}  ${value}
             END
          END
-         Spawn Notebook
+         Wait Until Keyword Succeeds    30    1    Spawn Notebook
```

Comment:
I think you need to specify what `30` is, otherwise it might be confusing.
You're also giving a 30s timeout to a keyword that has a 600s timeout by default: https://github.com/red-hat-data-services/ods-ci/blob/master/tests/Resources/Page/ODH/JupyterHub/JupyterHubSpawner.robot#L110

I think this is not needed, you can use the regular `Spawn Notebook` and increase the `retries` of this keyword if you think it can fail

Reply:
I disagree this 30 second is for the whole keyword to succeed Even if i try to increase the retires to the keyword it doesn't help the purpose because retry only come into the picture while checking the progress bar. This keyword will fail before that exactly at click button start server. I agree to add 's' for more clarity

Reply:
>This keyword will fail before that exactly at click button start server

I have never seen it fail when clicking that button, but it might fail due to JH not spawning a server (either timeout or other failure), but for that you already have the 600s timeout in the `Spawn Notebook` keyword itself. I don't understand the need for an additional timeout on top of it?

Reply:
In normal case it doesn't fail but when you reload page it take sometime to get reload if i don't give timeout, Start server button will not be available because page is still in reloading state. 

Coming back to 600s timeout if you look at the spawn keyword this 600s is only for the **id progress** which is the last line. If i don't give timeout it will fail in _start button_ (mentioned in italic)

Spawn Notebook
   [Documentation]  Start the notebook pod spawn and wait ${spawner_timeout} seconds (DEFAULT: 600s)
   [Arguments]  ${spawner_timeout}=600 seconds
   ${version-check} =  Is RHODS Version Greater Or Equal Than  1.5.0
   IF  ${version-check}==True
      _Click Button  Start Server_
      Wait Until Page Contains  Starting server
   ELSE
      _Click Button  Start server_
      Wait Until Page Contains  Your server is starting up
   END
   Wait Until Element is Visible  id:progress-bar
   **Wait Until Page Does Not Contain Element  id:progress-bar  ${spawner_timeout}**